app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -330,80 +334,40 @@ public void onCreate(Bundle savedInstanceState) {
     actionBar = getSupportActionBar();
-    RecyclerView tableDrawerRight = tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list);
+    tableDrawerRight = tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list);
     checkForRateDialog();
 
     initPlayStoreUri();
     isHideToolbar = sharedPreferenceUtil.getPrefHideToolbar();
 
-    FileReader fileReader = new FileReader();
-    documentParserJs = fileReader.readFile("js/documentParser.js", this);
-    documentSections = new ArrayList<>();
-    tabsAdapter = new TabsAdapter(this, webViewList);
-    tabsAdapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
-      @Override
-      public void onChanged() {
-        updateTabSwitcherIcon();
-      }
-    });
-
-    tableDrawerRight.setLayoutManager(new LinearLayoutManager(this));
-
-    TableDrawerAdapter tableDrawerAdapter = setupTableDrawerAdapter();
-    tableDrawerRight.setAdapter(tableDrawerAdapter);
-    tableDrawerAdapter.notifyDataSetChanged();
-
-    tabsAdapter.setTabClickListener(new TabsAdapter.TabClickListener() {
-      @Override
-      public void onSelectTab(View view, int position) {
-        hideTabSwitcher();
-        selectTab(position);
+    addFileReader();
+    setupTabsAdapter();
+    setTableDrawerInfo();
+    setTabListener();
 
-        /* Bug Fix
-         * Issue #592 in which the navigational history of the previously open tab (WebView) was
-         * carried forward to the newly selected/opened tab; causing erroneous enabling of
-         * navigational buttons.
-         */
-        updateBottomToolbarArrowsAlpha();
-      }
-
-      @Override
-      public void onCloseTab(View view, int position) {
-        closeTab(position);
-      }
-    });
     compatCallback = new CompatFindActionModeCallback(this);
     setUpTTS();
-    documentParser = new DocumentParser(new DocumentParser.SectionsListener() {
-      @Override
-      public void sectionsLoaded(String title, List<DocumentSection> sections) {
-        for (DocumentSection section : sections) {
-          if (section.title.contains("REPLACE_")) {
-            section.title = getResourceString(getApplicationContext(), section.title);
-          }
-        }
-        documentSections.addAll(sections);
-        if (title.contains("REPLACE_")) {
-          tableDrawerAdapter.setTitle(getResourceString(getApplicationContext(), title));
-        } else {
-          tableDrawerAdapter.setTitle(title);
-        }
-        tableDrawerAdapter.setSections(documentSections);
-        tableDrawerAdapter.notifyDataSetChanged();
-      }
 
-      @Override
-      public void clearSections() {
-        documentSections.clear();
-        tableDrawerAdapter.notifyDataSetChanged();
-      }
-    });
+    setupDocumentParser();
 
     manageExternalLaunchAndRestoringViewState();
     loadPrefs();
     updateTitle();
 
-    Intent i = getIntent();
+    i = getIntent();
+    setupIntent(i);
```

Comment:
You can pass `getIntent()` directly to `setupIntent` as `setupIntent(getIntent())` as it will reduce 2 statments.